File: FestivalJava/Services/src/main/java/ModelServices/Interfaces/IServiceArtist.java
package ModelServices.Interfaces;

import Domain.Artist;
import Validation.Exceptions.ValidatorException;

import java.util.List;

/**
 * Created by Sebi on 28-Mar-17.
 */
public interface IServiceArtist {
    /*
    Saves an artist
     */
    void save(Artist artist) throws ValidatorException;

    /*
    Deletes an artist
     */
    void delete(Integer idArtist);

    /*
    Updates an artist
     */
    void update(Integer id, Artist artist) throws ValidatorException;

    /*
    Returns an artist by id
     */
    Artist getById(Integer idArtist);

    /*
    Returns all artists
     */
    List<Artist> getAll();
}

File: FestivalC#/Festival/Services/Service/ModelServices/IServiceArtist.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Service
{
    public interface IServiceArtist
    {
        /*
         * Saves an artist 
         */
        void save(Artist artist);

        /*
         * Deletes an artist
         */
        void delete(int idArtist);

        /*
         * Updates an artist
         */
        void update(int id, Artist artist);

        /*
         * Gets an artist by id 
         * if not found return null
         */
        Artist getById(int idArtist);

        /*
         * Returns a list of artists
         */
        List<Artist> getAll();
    }
}
